src/lib/clients/phony.ts

```diff
@@ -1,21 +1,69 @@
-import { BaseOptions } from "../../types/generic";
-import { Client } from "../../types/generic";
+import { BaseOptions, HandlerBody } from '../../types/generic';
+import { SayArguments } from "@slack/bolt";
+import { KeeperSlackMiddleware } from "../../types/clients/keeper"
+import { CommandClient, Command } from '../../types/generic';
 import { Logger } from '../../types/logger'
+import { EventEmitter } from 'events';
+import * as net from 'net';
 
-export default class PhonyClient implements Client {
+export default class PhonyClient extends EventEmitter implements CommandClient<PhonyClient> {
     private logger: Logger;
+    private server: net.Server;
+    private commands: Command<KeeperSlackMiddleware, this>[];
 
-    constructor (options: BaseOptions) {
+    constructor(options: BaseOptions) {
+        super();
+        this.commands = [];
         this.logger = options.logger;
+        this.server = net.createServer(c => {
+            c.on('connect', () => this.logger.log('tcp connection opened'));
+            c.on('end', () => this.logger.log('tcp connection closed'));
+
+            c.on('data', async (data) => {
+                const msg = data.toString().trim();
+
+                await Promise.all(this.commands.filter(({ matches }) => {
+                    switch (typeof matches) {
+                        case "object": return matches.test(msg);
+                        case "string": return matches == msg;
+                    }
+                }).map(({ handler }) => {
+                    return handler(this.handlerArgs(c, msg))
+                }));
+            });
+        });
+    }
+
+    handlerArgs(c: net.Socket, text: string): HandlerBody<KeeperSlackMiddleware, this> {
+        return {
+            client: this,
+            logger: this.logger.log,
+            command: {
+                say: async (args: string | SayArguments) => {
+                    const text = typeof args === "string" ? args : args.text;
+                    c.write(`keeper: ${text}\n`);
+                    return { ok: true };
+                },
+                payload: {
+                    channel: "#local",
```

Comment:
No need to add this in this PR, but: Could we create some sort of helper or factory to generate this data? Just thinking eventually when tests get written, we'll need such a function, which could be helpful here as well

Reply:
Yeah I could see that being useful! Let's cross that bridge when we get to it though. I tried keeping the payload we work with [small by limiting the properties](https://github.com/devict/keeper/blob/local-client/src/types/clients/keeper.ts#L15) we actually use, and if it stays small it might not be a big deal to just specify these three properties.

Either way, if it becomes a burden when tests are being written I'm sure we'll have a better idea of what that should look like and where it should live.